notification/irc.py

```diff
@@ -191,6 +198,7 @@ def main():
             server=dict(default='localhost'),
             port=dict(default=6667),
             nick=dict(default='ansible'),
+            nick_to=dict(),
```

Comment:
I'm thinking about the relationship between channel and nick_to here.  I think that in most cases the user will want to either send to a channel or to one or more irc nicks.  A few might want to send to both a channel and one or more IRC nicks. So perhaps the following would be appropriate:

* Make channel optional
* Use required_one_of() to make either nick_to or channel be set.
* Check that if topic is set, then channel is set.  (I don't think any of the pre-existing AnsibleModule checks will handle this so you'll just have to test it below when you get the values out of module.params)
* Make nick_to a list so that more than one irc nick can be specified

(And then update documentation and code appropriately to note the new behaviour and when it changed).